src/infrasys/time_series_metadata_store.py

```diff
@@ -5,57 +5,108 @@
 import json
 import sqlite3
 from dataclasses import dataclass
+from pathlib import Path
 from typing import Any, Iterable, Optional, Sequence
 from uuid import UUID
 
 from loguru import logger
 
-from infrasys.exceptions import ISAlreadyAttached, ISOperationNotAllowed, ISNotStored
-from infrasys import Component
-from infrasys.supplemental_attribute_manager import SupplementalAttribute
+from infrasys import (
+    KEY_VALUE_STORE_TABLE,
+    TIME_SERIES_ASSOCIATIONS_TABLE,
+    TIME_SERIES_METADATA_TABLE,
+    Component,
+    __version__,
+)
+from infrasys.exceptions import ISAlreadyAttached, ISNotStored, ISOperationNotAllowed
 from infrasys.serialization import (
+    TYPE_METADATA,
+    SerializedTypeMetadata,
     deserialize_value,
     serialize_value,
-    SerializedTypeMetadata,
-    TYPE_METADATA,
 )
+from infrasys.supplemental_attribute_manager import SupplementalAttribute
 from infrasys.time_series_models import (
-    TimeSeriesMetadata,
-    SingleTimeSeriesMetadataBase,
     NonSequentialTimeSeriesMetadataBase,
+    SingleTimeSeriesMetadataBase,
+    TimeSeriesMetadata,
 )
 from infrasys.utils.sqlite import execute
+from infrasys.utils.time_utils import to_iso_8601
 
 
 class TimeSeriesMetadataStore:
     """Stores time series metadata in a SQLite database."""
 
-    TABLE_NAME = "time_series_metadata"
-
     def __init__(self, con: sqlite3.Connection, initialize: bool = True):
         self._con = con
         if initialize:
-            self._create_metadata_table()
+            self._create_associations_table()
+            self._create_key_value_store()
+        self._cache_metadata: dict[UUID, TimeSeriesMetadata] = {}
+
+    def _load_metadata_into_memory(self):
+        query = f"SELECT json(metadata) FROM {TIME_SERIES_METADATA_TABLE}"
+        cursor = self._con.cursor()
+        cursor.execute(query)
+        rows = cursor.fetchall()
+        for row in rows:
+            metadata = _deserialize_time_series_metadata(row[0])
+            self._cache_metadata[metadata.uuid] = metadata
+        cursor.execute(f"DROP TABLE {TIME_SERIES_METADATA_TABLE}")
+        self._con.commit()
+
+    def _create_key_value_store(self):
+        schema = ["key TEXT PRIMARY KEY", "value JSON NOT NULL"]
+        schema_text = ",".join(schema)
+        cur = self._con.cursor()
+        execute(cur, f"CREATE TABLE {KEY_VALUE_STORE_TABLE}({schema_text})")
+        self._create_indexes(cur)
+
+        rows = [("infrasys_version", __version__)]
```

Comment:
This version needs to be something that both InfrastructureSystems.jl and infrasys understand. I put `1.0.0` in my Julia branch. I'm proposing that the version governs the schema of this database and not the software versions of the packages.

Reply:
Just changed it to 1.0.0

Reply:
And matched IS.jl